### ISSUE TITLE
Error while fine-tuning

### ISSUE THREAD

**Issue by Eniac-Xie** [yjxiong/caffe#184]
Hi, @yjxiong , thanks for your code!

I replace the BatchNorm and Scale layer in Kaiming's ResNet50 with your SyncBN, and get a new net (which we can call it ResNet-SyncBN).

I write a script to initialize ResNet-SyncBN with the original ResNet50 , and test it on Imagenet val dataset, get a reasonable result.

But if I use the ResNet50-SyncBN (initialized with Kaiming's ResNet50) for fine-tuning, the loss always become 87.3365 at the second iteration, and the parameter become nan (I tried to use small learning rate, it does not help).

And if I just random initialize the ResNet50-SyncBN, the loss seems reasonable.

Have you ever fine-tuning the SyncBN layer?

the code I used to initial ResNet50-SyncBN with ResNet50 is:
`
import caffe

def main():
    net_kaiming = caffe.Net('resnet50/resnet50_kaiming.prototxt', 'resnet50/resnet50_kaiming.caffemodel', caffe.TEST)
    net_sync_bn = caffe.Net('resnet50/resnet50_sync_bn.prototxt', 'resnet50/resnet50_kaiming.caffemodel', caffe.TEST)

    for name, param in net_sync_bn.params.iteritems():
        if 'sync_bn' in name:
            kaiming_bn_name = name.replace('sync_bn', 'bn')
            kaiming_scale_name = name.replace('sync_bn', 'scale')
            print 'init: ', name
            net_sync_bn.params[name][0].data[...] = net_kaiming.params[kaiming_scale_name][0].data # init weight
            net_sync_bn.params[name][1].data[...] = net_kaiming.params[kaiming_scale_name][1].data # init bias
            net_sync_bn.params[name][2].data[...] = net_kaiming.params[kaiming_bn_name][0].data / net_kaiming.params[kaiming_bn_name][2].data # init running mean
            net_sync_bn.params[name][3].data[...] = net_kaiming.params[kaiming_bn_name][1].data / net_kaiming.params[kaiming_bn_name][2].data # init running var
    net_sync_bn.save('resnet50/resnet50_sync_bn.caffemodel')

`

and I used SyncBN in prototxt as below:

layer {
  name: "sync_bn_conv1"
  type: "SyncBN"
  bottom: "conv1"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  param {
    lr_mult: 0
    decay_mult: 0
  }
  param {
    lr_mult: 0
    decay_mult: 0
  }
  bn_param {
    slope_filler {
      type: "constant"
      value: 1
    }    
    bias_filler {
      type: "constant"
      value: 0
    }    
    momentum: 0.9997
    eps: 1e-05
    frozen: false
  }
}

**Comment by yjxiong** [comment]
```
bottom: "conv1"
top: "conv1"
```

I don't think SyncBN supports in-place operation. 

BTW, your momentum set to 0.9997 seems very strange. What is the task you are fine-tuning on?

**Comment by Eniac-Xie** [comment]
Thanks for your response! @yjxiong 
I use 0.9997 as Deeplab v3 did. I want to re-produce Deeplab v3 with caffe.

When I use SyncBN as no-inplace operation, the network can converge, thank you!

**Comment by yjxiong** [comment]
Good to know. 

Maybe it is worth to note that using this large momentum is mostly like freezing the statistics of the SyncBN layers and only tuning the scale and bias parameters (these two are updated via SGD). In this case, SyncBN is not necessary.

**Comment by John1231983** [comment]
@Eniac-Xie : I also reproduce the deeplab v3 performance. However, I cannot achieve his performance (I reproduce with the batch size of 8). I only achieve **73.6**%, while the report is **75.78**% with resnet pretrained model in tensorflow. Do you obtain a good performance? Do you have some tips to obtain it?
